src/dscanner/analysis/redundant_storage_class.d

```diff
@@ -5,68 +5,106 @@
 
 module dscanner.analysis.redundant_storage_class;
 
-import std.stdio;
-import std.string;
-import dparse.ast;
-import dparse.lexer;
 import dscanner.analysis.base;
 import dscanner.analysis.helpers;
-import dsymbol.scope_ : Scope;
+import std.format;
+import std.algorithm.comparison : among;
+import std.algorithm.searching: all;
+import dmd.astenums : STC;
 
 /**
  * Checks for redundant storage classes such immutable and __gshared, static and __gshared
  */
-final class RedundantStorageClassCheck : BaseAnalyzer
+extern(C++) class RedundantStorageClassCheck(AST) : BaseAnalyzerDmd
 {
-	alias visit = BaseAnalyzer.visit;
-	enum string REDUNDANT_VARIABLE_ATTRIBUTES = "Variable declaration for `%s` has redundant attributes (%-(`%s`%|, %)).";
 	mixin AnalyzerInfo!"redundant_storage_classes";
+	alias visit = BaseAnalyzerDmd.visit;
 
-	this(string fileName, bool skipTests = false)
+	extern(D) this(string fileName, bool skipTests = false)
 	{
-		super(fileName, null, skipTests);
+		super(fileName, skipTests);
 	}
 
-	override void visit(const Declaration node)
+	override void visit(AST.VarDeclaration d)
```

Comment:
Note that this check is for a Declaration type (which in dmd is an ancestor of VarDeclaration). This means that the original check would verify all kinds of declarations (Vars, Funcs etc.), whereas you are restricting it just to var declarations. I don't know if this was the case for dscanner, but you should check it up.